cmake/std/atdm/utils/get_known_system_info.sh

```diff
@@ -11,127 +11,196 @@ unset ATDM_CONFIG_CDASH_HOSTNAME
+unset systemNameTypeMatchedListHostNames
+declare -A systemNameTypeMatchedListHostNames
 
-# environments available on rhel systems
-elif [[ $ATDM_CONFIG_BUILD_NAME == *"spack-rhel"* ]] ; then
-  ATDM_HOSTNAME=spack-rhel
-  ATDM_SYSTEM_NAME=spack-rhel
-elif [[ -f /projects/sems/modulefiles/utils/get-platform ]] ; then
+# TLCC2 systems
+if [[ $SNLSYSTEM == "tlcc2"* ]] ; then
+  systemNameTypeMatchedList+=(tlcc2)
+  systemNameTypeMatchedListHostNames[tlcc2]=$SNLCLUSTER
+fi
+
+# SEMS and CEE RHEL systems
+if [[ -f /projects/sems/modulefiles/utils/get-platform ]] ; then
   # This machine has the SEMS modules!
-  ATDM_SYSTEM_NAME=`source /projects/sems/modulefiles/utils/get-platform`
-  if [[ $ATDM_SYSTEM_NAME == "rhel7-x86_64" ]] ; then
-    # This is a RHEL7 platform that has the SEMS modules.
-    ATDM_HOSTNAME=sems-rhel7
-    ATDM_SYSTEM_NAME=sems-rhel7
-  elif [[ $ATDM_SYSTEM_NAME == "rhel6-x86_64" ]] ; then
-    # This is a RHEL6 platform that has the SEMS modules.  But is this also a
-    # CEE LAN mahcine?
-    if [[ -f /projects/sparc/modules/cee-rhel6/sparc/master ]] ; then
-      ATDM_IS_CEE_RHEL6_MACHINE=1
+  ATDM_SEMS_SYSTEM_NAME=`source /projects/sems/modulefiles/utils/get-platform`
+  if [[ "${ATDM_SEMS_SYSTEM_NAME}" == "rhel6-x86_64" ]] ; then
+    systemNameTypeMatchedList+=(sems-rhel6)
+    systemNameTypeMatchedListHostNames[sems-rhel6]=sems-rhel6
+    if [[ -d /projects/sparc/tpls ]] ; then
+      systemNameTypeMatchedList+=(cee-rhel6)
+      systemNameTypeMatchedListHostNames[cee-rhel6]=cee-rhel6
     fi
-    # Now select the env based on the above logic
-    if [[ $ATDM_CONFIG_BUILD_NAME == *"sems-rhel6"* ]] ; then
-      ATDM_HOSTNAME=sems-rhel6
-      ATDM_SYSTEM_NAME=sems-rhel6
-    elif [[ $ATDM_CONFIG_BUILD_NAME == *"cee-rhel6"* ]] ; then
-      if [[ $ATDM_IS_CEE_RHEL6_MACHINE == "1" ]] ; then
-        # This is a CEE RHEL6 machine and 'cee-rhel6' was given in build name,
-        # so select the system name 'sem-rhel6'
-        ATDM_SYSTEM_NAME=cee-rhel6
-        ATDM_HOSTNAME=cee-rhel6
-      else
-        echo
-        echo "***"
-        echo "*** Error, hostname='$ATDM_CONFIG_REAL_HOSTNAME' is a 'sems-rhel6' machine but is"
-        echo "*** is *not* a 'cee-rhel6' machine but 'cee-rhel6' was given in the"
-        echo "*** build name string '$ATDM_CONFIG_BUILD_NAME'!  Please remove 'cee-rhel6'"
-	echo "*** from the build name or provide 'sems-rhel6' and then the 'sems-rhel6'"
-	echo "*** env will be used."
-        echo "***"
-        return
-      fi
-    else
-      # 'cee-rhel6' nor 'sems-rhel6' was set in the build name, so the default
-      # is to sue the 'sems-rhel6' env!
-      ATDM_HOSTNAME=sems-rhel6
-      ATDM_SYSTEM_NAME=sems-rhel6
+  elif [[ "${ATDM_SEMS_SYSTEM_NAME}" == "rhel7-x86_64" ]] ; then
+    systemNameTypeMatchedList+=(sems-rhel7)
+    systemNameTypeMatchedListHostNames[sems-rhel7]=sems-rhel7
+    if [[ -d /projects/sparc/tpls ]] ; then
```

Comment:
Same as above. Does the existence of this directory guarantee that we're on a cee-rhel6 system?

Reply:
Will be fixed as per above.